### PR TITLE
add last completed to appointment

### PR DESCRIPTION
**Pull request by ritabiniash**
On appointment completion, add/update lastCompletedDonationTime to donor.

### CODE REVIEW

File: packages/functions/src/donor/CompleteAppointmentHandler.ts [hunk 1]
```diff
@@ -60,6 +65,25 @@ export async function completeAppointmentFunc(
 
   const completedAppointment =
     await DbAppointmentUtils.toBookedAppointmentAsync(updatedAppointment);
+  
+  const currentAppointmentDonorId = appointment.donorId;
+
+  if (updatedAppointment.status == AppointmentStatus.COMPLETED && 
+    currentAppointmentDonorId != MANUAL_DONOR_ID) {
```
Comment (yaron1m) [inline packages/functions/src/donor/CompleteAppointmentHandler.ts:72]: Consider using `AppointmentUtils.isManualDonor`

File: packages/functions/src/dal/DonorDataAccessLayer.ts [hunk 2]
```diff
@@ -79,3 +79,14 @@ export async function getDonorsByGroupIds(
     .get();
   return snapshot.docs.map<DbDonor>((doc) => doc.data() as DbDonor);
 }
+
+export async function updateDonor(donorId: string, 
+  fieldsToUpdate: Partial<DbDonor>) {
+    const collection = admin.firestore().collection(Collections.DONORS);
+    const donorRef = collection.doc(donorId);
+    const donorDoc = await donorRef.get()
```
[the remark this answers is not in the document]
Comment (ritabiniash) [reply packages/functions/src/dal/DonorDataAccessLayer.ts:87]: 1. This is the firebase way to check if a document exists https://stackoverflow.com/questions/47308159/whats-the-best-way-to-check-if-a-firestore-record-exists-if-its-path-is-known
2.It won't throw an error if it doesn't exist

[the remark this answers is not in the document]
Comment (yaron1m) [reply packages/functions/src/dal/DonorDataAccessLayer.ts]: Then why do we need this check?
Either the document exists and we update it, or it doesn't and no error is thrown.
So it seems to me like the code will behave the same as it does today even if we remove this fetch, right?

[the remark this answers is not in the document]
Comment (ritabiniash) [reply packages/functions/src/dal/DonorDataAccessLayer.ts]: will check

[the remark this answers is not in the document]
Comment (ritabiniash) [reply packages/functions/src/dal/DonorDataAccessLayer.ts]: ok so it does throw an error if it doesn't exist
I wonder what is the best practice here, should this function raise an error if the Donor doesn't exist ?

[the remark this answers is not in the document]
Comment (yaron1m) [reply packages/functions/src/dal/DonorDataAccessLayer.ts]: I am ok with it

[the remark this answers is not in the document]
Comment (yaron1m) [reply packages/functions/src/dal/DonorDataAccessLayer.ts]: Let's remove the existence check, and I think this PR will be ready

**Review by yaron1m — APPROVED** [review]
Great work!

**Pull request merged**